### PR TITLE
Added dynamic delimiter and maxDataLength feature.

### PR DESCRIPTION
**Pull request by l0oky**
Corrected version of the previous pull request. Had to make a new PR because I accidentally deleted my `actionhero` fork. Silly me.

This version has better config names and overall I followed your comments, any new comments are appriciated :)

### CODE REVIEW

File: servers/socket.js [hunk 1]
```diff
@@ -107,7 +107,14 @@ var initialize = function(api, options, next){
...
     var parseLine = function(line){
-      if(line.length > 0){
+      var blen = Buffer.byteLength(line, 'utf8');
+      if(blen > api.config.servers.socket.maxDataLength && api.config.servers.socket.maxDataLength !== 0){
+        var error = api.config.errors.dataLengthTooLarge(api.config.servers.socket.maxDataLength,blen);
+        server.log(error, 'error');
+        server.sendMessage(connection, {status:'error', error: error, context: 'response'});
+        return;
+      }
+      if(blen > 0){
```
Comment (l0oky) [inline servers/socket.js]: This shouldn't be blen. I am fixing this right now.

File: servers/socket.js [hunk 2]
```diff
@@ -107,14 +107,16 @@ var initialize = function(api, options, next){
...
-        var error = api.config.errors.dataLengthTooLarge(api.config.servers.socket.maxDataLength,blen);
-        server.log(error, 'error');
-        server.sendMessage(connection, {status:'error', error: error, context: 'response'});
-        return;
+      if(api.config.servers.socket.maxDataLength !== 0){
+        var blen = Buffer.byteLength(line, 'utf8');
+        if( blen > api.config.servers.socket.maxDataLength){
+          var error = api.config.errors.dataLengthTooLarge(api.config.servers.socket.maxDataLength,blen);
+          server.log(error, 'error');
+          server.sendMessage(connection, {status:'error', error: error, context: 'response'});
```
Comment (l0oky) [inline servers/socket.js]: Still not sure about this message format. Is this the standard?

File: servers/socket.js [hunk 3]
```diff
@@ -112,8 +112,7 @@ var initialize = function(api, options, next){
         if( blen > api.config.servers.socket.maxDataLength){
           var error = api.config.errors.dataLengthTooLarge(api.config.servers.socket.maxDataLength,blen);
           server.log(error, 'error');
-          server.sendMessage(connection, {status:'error', error: error, context: 'response'});
-          return;
+          return server.sendMessage(connection, {status:'error', error: error, context: 'response'});
```
Comment (l0oky) [inline servers/socket.js]: Still not sure about this message format. Is this the standard?

`{status:'error', error: error, context: 'response'}`

Comment (l0oky) [inline servers/socket.js]: Still not sure about this message format. Is this the standard?

`{status:'error', error: error, context: 'response'}`

File: servers/socket.js [hunk 4]
```diff
@@ -107,6 +107,14 @@ var initialize = function(api, options, next){
     connection.params = {};
 
     var parseLine = function(line){
+      if(api.config.servers.socket.maxDataLength !== 0){
```
Comment (evantahler) [inline servers/socket.js]: probably `> 0`

**Comment by evantahler** [comment]
nice!
change out that one `>` and this is good to go

**Comment by l0oky** [comment]
There we go :) Thanks for your help, getting familiar with the `actionhero` core day by day.

**Pull request merged**